spark/v3.4/spark/src/main/java/org/apache/iceberg/spark/data/vectorized/VectorizedSparkParquetReaders.java

```diff
@@ -67,7 +70,17 @@ public static ColumnarBatchReader buildReader(
                 NullCheckingForGet.NULL_CHECKING_ENABLED,
                 idToConstant,
                 ColumnarBatchReader::new,
-                deleteFilter));
+                deleteFilter,
+                bufferAllocator));
+  }
+
+  public static ColumnarBatchReader buildReader(
+      Schema expectedSchema,
+      MessageType fileSchema,
+      Map<Integer, ?> idToConstant,
+      DeleteFilter<InternalRow> deleteFilter) {
+    return buildReader(
+        expectedSchema, fileSchema, idToConstant, deleteFilter, ArrowAllocation.rootAllocator());
```

Comment:
do we also need to change the `buildCometReader` below?

Reply:
I didn't do anything to test the comet integration yet, I think we can hold off on that mod until we do a comet followup (if we need one?)